packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/glue/start-job-run.ts

```diff
@@ -42,6 +42,21 @@ export interface GlueStartJobRunProps extends sfn.TaskStateBaseProps {
    * @default - Default delay set in the job definition
    */
   readonly notifyDelayAfter?: Duration;
+
+  /**
+   * The type of predefined worker that is allocated when a job runs.
+   * This must be set along with numberOfWorkers.
+   *
+   * @default - differs based on specific Glue version
+   */
+  readonly workerType?: WorkerType;
+
+  /**
+   * The number of workers of a defined `WorkerType` that are allocated when a job runs.
+   *
+   * @default - differs based on specific Glue version/worker type
+   */
+  readonly numberOfWorkers?: number;
```

Comment:
For my own understanding, the original issue is asking for `AllocatedCapacity` which was deprecated in favor of `MaxCapacity`. How does adding these 2 properties solve the original issue?

Reply:
@aaythapa 
Thank you for the review. Apologies for the insufficient explanation. I have also added an explanation to the pull request.

MaxCapacity can only be used with Glue version 1 and earlier, as version 2 and later require the use of Worker type and the Number of workers.

https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-jobs-runs.html#aws-glue-api-jobs-runs-StartJobRun


> MaxCapacity – Number (double).
For Glue version 1.0 or earlier jobs, using the standard worker type, the number of AWS Glue data processing units (DPUs) that can be allocated when this job runs. A DPU is a relative measure of processing power that consists of 4 vCPUs of compute capacity and 16 GB of memory. For more information, see the AWS Glue pricing page.
> For Glue version 2.0+ jobs, you cannot specify a Maximum capacity. Instead, you should specify a Worker type and the Number of workers.
Do not set MaxCapacity if using WorkerType and NumberOfWorkers.

Furthermore, Glue version 1 and earlier have already reached EOS.

https://docs.aws.amazon.com/glue/latest/dg/glue-version-support-policy.html

Therefore, instead of adding MaxCapacity, I deemed it appropriate to add Worker type and the Number of workers, as indicated in the provided documentation.

Reply:
Nice, thank you for the explanation.